File: trivia/scala/com/adaptionsoft/games/trivia/runner/GameRunner.scala
package com.adaptionsoft.games.trivia.runner

import com.adaptionsoft.games.uglytrivia.Game
import java.util.Random


object GameRunner {
  var notAWinner = false

  def main(args: Array[String]) {
    var aGame = new Game();
    aGame.add("Chet")
    aGame.add("Pat")
    aGame.add("Sue")

    var rand: Random = new Random

    do {
      aGame.roll(rand.nextInt(5) + 1)
      if (rand.nextInt(9) == 7) {
        notAWinner = aGame.wrongAnswer
      }
      else {
        notAWinner = aGame.wasCorrectlyAnswered
      }
    } while (notAWinner)
  }
}

File: trivia/java/src/main/java/trivia/GameRunner.java

package trivia;

import java.util.Random;


public class GameRunner {


    public static void main(String[] args) {
        boolean notAWinner = false;
        Game aGame = new Game();

        aGame.startWithPlayers("Chet", "Pat", "Sue");

        Random rand = new Random();

        do {

            aGame.nextTurn();
            if (aGame.roll(rand.nextInt(5) + 1)) {

                if (rand.nextInt(5) == 4) {
                    notAWinner = aGame.wrongAnswer();
                } else {
                    notAWinner = aGame.wasCorrectlyAnswered();
                }
            }


        } while (notAWinner);

    }
}
